File: management/vo/workspace.go
package vo

type WorkspaceVo struct {
	ID string `json:"id"`

	// slug对应前端输入的想要的空间
	Slug string `json:"name"` // URL标识，如 "tencent-rd"

	//对应k8s的 namespace
	Namespace string `json:"namespace"`

	// 显示名称：用户在 Vercel 风格界面看到的名称 (如 "我的私有云")
	DisplayName string `json:"displayName"`

	TokenCount int64 `json:"tokenCount"`

	QuotaUsage int64 `json:"quotaUsage"`

	NodeCount int64 `json:"nodeCount"`

	// 状态
	Status string `json:"status"` // active, terminating, frozen

	Members []UserVo `json:"members,omitempty"`
}

File: fronted/src/api/workspace.ts
import request from '@/api/request';

export interface Workspace {
    id: string;
    slug: string;
    namespace?: string;
    displayName: string;
    nodeCount: number;
    tokenCount: number;
    maxNodeCount: number;
    status: 'active' | 'inactive';
    createdAt: string;
    createdBy?: string;
    // 网络信息
    networkName?: string;
    networkCIDR?: string;
    networkStatus?: string;
    // 配额使用量
    quotaUsage?: number;
    // 成员列表
    members?: Array<{ id: string; name: string; role: string }>;
}

// --- Workspace空间管理 ---
export const add    = (data?: any)              => request.post('/workspaces/add', data);
export interface ListWsParams {
    page?: number
    pageSize?: number
    search?: string
    status?: string
}
export const listWs = (params?: ListWsParams) => request.get('/workspaces/list', params);
export const updateWs = (id: string, data?: any) => request.put(`/workspaces/${id}`, data);
export const deleteWs = (id: string)            => request.delete(`/workspaces/${id}`);

